internal/controller/policyserver_controller_deployment.go

```diff
@@ -168,6 +168,7 @@ func (r *PolicyServerReconciler) updatePolicyServerDeployment(policyServer *poli
 				SecurityContext:    podSecurityContext,
 				Containers:         []corev1.Container{admissionContainer},
 				ServiceAccountName: policyServer.Spec.ServiceAccountName,
+				Tolerations:        policyServer.Spec.Tolerations,
```

Comment:
Just in case, shouldn't it be done with something similar to https://github.com/kubewarden/kubewarden-controller/blob/main/internal/controller/policyserver_controller_deployment.go#L299-L300?

That way, we support the users changing the Deployment if there was no PolicyServer spec.tolerations set.

Reply:
@viccuad  I don't understand why we wrote the code you linked. I would have handled affinity like @jvanz just did here with tolerations...

@fabriziosestito can you chime in?

Reply:
Is not related to how we used to detect changes in the resource and trigger a reconciliation? I have a vague memory about a discussion around this. However, as now we use the controller-runtime helper functions, I believe, we do not need to do that anymore.

Reply:
We can remove it and use the same approach of the Tolerations field. As far as I  understood the `if` block does not change the reconciliation behaviour, it's probably some leftover since now we use the controller-runtime `CreatOr*`
> That way, we support the users changing the Deployment if there was no PolicyServer spec.tolerations set.
I don't think the user should be able to thinker with the Deployment, as the only interface we expose is the PolicyServer CRD

Reply:
If we agree then, I would like to follow the same approach of Tolerations for Affinity in this PR.